x-pack/lib/monitoring/inputs/metrics.rb

```diff
@@ -211,9 +211,18 @@ def fetch_global_stats
     end
 
     def extract_cluster_uuids(stats)
-      result = stats.extract_metrics([:stats, :pipelines, :main, :config], :cluster_uuids)
-      if result && !result[:cluster_uuids].empty?
-        cluster_uuids = result[:cluster_uuids]
+      cluster_uuids = []
+      agent.running_pipelines.each do |pipeline_id, _|
+        unless pipeline_id.to_sym == :".monitoring-logstash"
```

Comment:
the yielded `Pipeline` responds to `Pipeline#system?`, which is true for the monitoring pipeline -- would using it be safer than hard-coding a skip pipeline?

Secondly for readability I would prefer to special-case the skip clause as `next if ...` instead of providing a block to unless, since unless blocks are inherently harder to track, e.g., 

~~~ ruby
agent.running_pipelines.each do |pipeline_id, pipeline|
  next if pipeline.system?
  # ...
end
~~~